File: latihan1_b/c++/DataTim.cpp
#include <iostream>
#include <string>

using namespace std;

class DataTim
{
    private:
        /* atribut private data Tim Sepakbola */
        string namaTim;
        string negaraAsal;
        int tahunBerdiri;

        // objek yang menyimpan data pemain
        class DataPemain
        {
            private:
                /* atribut private */
                string namaPemain[50];
                int jumlahPemain;
            public:
                DataPemain(){}

                DataPemain(int pointer, string np){
                    namaPemain[pointer] = np;
                }

                void setNamaPemain(int pointer, string np){
                    namaPemain[pointer] = np;
                }

                string getNamaPemain(int pointer){
                    return namaPemain[pointer];
                }

                void setJumlah(int jp){
                    jumlahPemain = jp;
                }

                int getJumlah(){
                    return jumlahPemain;
                }

                void showPemain(){
                    int i;
                    for (i = 0; i < jumlahPemain; i++)
                    {
                        /* code */
                        cout << "   " << (i+1) << ". " << namaPemain[i] << endl;
                    }
                }

                ~DataPemain(){}
        };

        // deklarasi objek dataPemain
        DataPemain listPemain;

    public:
        /* constructor */
        DataTim() {
        }

        /* constructor yang menggunakan parameter */
        DataTim(string nt, string na, int tb){
            /* mengisi class dataTim*/
            namaTim = nt;
            negaraAsal = na;
            tahunBerdiri = tb;
        }

        /* membuat getter dan setter untuk setiap variabel */

        void setNamaTim(string nt){
            namaTim = nt;
        }

        string getNamaTim(){
            return namaTim;
        }

        void setNegaraAsal(string na){
            negaraAsal = na;
        }

        string getNegaraAsal(){
            return negaraAsal;
        }

        void setTahunBerdiri(int tb){
            tahunBerdiri = tb;
        }

        int getTahunBerdiri(){
            return tahunBerdiri;
        }

        void setPemain(int pointer, string np){
            listPemain.setNamaPemain(pointer, np);
        }

        void setJumlahPemain(int jp){
            listPemain.setJumlah(jp);
        }

        void showTim(){
            cout << "\n>> " << namaTim << " <<" << endl;
            cout << "Negara Asal : " << negaraAsal << endl;
            cout << "Tahun Berdiri : " << tahunBerdiri << endl;
            cout << "Daftar Pemain :" << endl;
            listPemain.showPemain(); 
        }

        /* destructor */
        ~DataTim(){}
};

File: latihan1_b/java/DataTim.java
public class DataTim {

    private String namaTim;
    private String negaraAsal;
    private int tahunBerdiri;

    private class DataPemain {
    
        private String namaPemain[] = new String[50];
        private int jumlah;

        public DataPemain(){
        }

        public void setNamaPemain(int pointer, String np) {
            this.namaPemain[pointer] = np;
        }

        public String getNamaPemain(int pointer) {
            return namaPemain[pointer];
        }

        public void setJumlah(int jp) {
            this.jumlah = jp;
        }

        public int getJumlah() {
            return jumlah;
        }

        public void showPemain() {
            int i = 0;
            for (i = 0; i < jumlah; i++) {
                System.out.println("   " + (i+1) + ". " + namaPemain[i]);
            }
        }
    }

    private DataPemain listPemain;

    /* konstruktor */
    public DataTim(){
        listPemain = new DataPemain();
    }
    public DataTim(String nt, String na, int tb){
        listPemain = new DataPemain();
        this.namaTim = nt;
        this.negaraAsal = na;
        this.tahunBerdiri = tb;
    }

    /* buat set dan get untuk seluruh variabel */

    public void setNamaTim(String nt) {
        this.namaTim = nt;
    }

    public String getNamaTim() {
        return namaTim;
    }

    public void setNegaraAsal(String na) {
        this.negaraAsal = na;
    }

    public String getNegaraAsal() {
        return negaraAsal;
    }

    public void setTahunBerdiri(int tb) {
        this.tahunBerdiri = tb;
    }

    public int getTahunBerdiri() {
        return tahunBerdiri;
    }

    public void setPemain(int pointer, String np) {
        listPemain.setNamaPemain(pointer, np);
    }

    public void setJumlahPemain(int jp) {
        listPemain.setJumlah(jp);
    }
    
    public void showTim() {
        System.out.println("\n>> " + namaTim + " <<");
        System.out.println("Negara Asal   : " + negaraAsal);
        System.out.println("Tahun Berdiri : " + tahunBerdiri);
        System.out.println("Daftar Pemain :");
        listPemain.showPemain();
    }
}
